store/v2/snapshots/manager.go

```diff
@@ -413,8 +412,9 @@ func (m *Manager) doRestoreSnapshot(snapshot types.Snapshot, chChunks <-chan io.
 
 	nextItem, err = m.commitSnapshotter.Restore(snapshot.Height, snapshot.Format, streamReader, chStorage)
 	if err != nil {
-		return errorsmod.Wrap(err, "multistore restore")
+		return err
 	}
+	close(chStorage)
```

Comment:
Lets add a comment so it doesn't get linted away in a future refactor

Reply:
It's hard to present it as a unit test, but u can try this [scripts](https://github.com/cosmos/cosmos-sdk/blob/hieu/snapshot-cmd/scripts/test_snapshots_export_restore.sh)

My understanding is in `storage.Restore` we loop through the channel:
```go
for kvPair := range chStorage {
```
This loop never ends until the channel is closed. 

```go
chStorage := make(chan *corestore.StateChanges, defaultStorageChannelBufferSize)
defer close(chStorage)

storageErrs := make(chan error, 1)
go func() {
  defer close(storageErrs)
  err := m.storageSnapshotter.Restore(snapshot.Height, chStorage)
  if err != nil {
	  storageErrs <- err
  }
}()

nextItem, err = m.commitSnapshotter.Restore(snapshot.Height, snapshot.Format, streamReader, chStorage)

...

// wait for storage snapshotter to complete
if err := <-storageErrs; err != nil {
  return errorsmod.Wrap(err, "storage snapshotter")
}
```

With our current implement, `close(chStorage)` not be called => loop never ends

Reply:
cc @cool-develope

Reply:
i think its because the `mockStorageSnapshotter` we use in restore test doesn't do anything, let me update that in my test PR